menagerie/pattern-predicate-protocol/tools/lift_kernel.py

```
@@ -0,0 +1,147 @@
+
+Usage: lift_kernel.py <kernel_root> <subdirs...> <substrate.db>
+Example: lift_kernel.py /tmp/linux-source/linux net/ipv4 net/ipv6 net/rxrpc /tmp/substrate-barrage/substrate.db
+"""
+import json
+import pathlib
+import sqlite3
+import subprocess
+import sys
+import time
+
+LIFTER = "/tmp/provekit-substrate/implementations/c/provekit-lift-c-kernel-doc/provekit-lift-c-kernel-doc"
```

Ruff (0.15.12)

[error] Probable insecure usage of temporary file or directory: "/tmp/provekit-substrate/implementations/c/provekit-lift-c-kernel-doc/provekit-lift-c-kernel-doc"

(S108)